dali/base/dadfs.cpp

```diff
@@ -2738,6 +2738,20 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
         return *t;
     }
 
+    IPropertyTree &queryHistory()
+    {
+        IPropertyTree *t = root->queryPropTree("History");
+        if (!t)
+            t = root->setPropTree("History",createPTree("History")); // takes ownership
```

Comment:
My first problem is: The every query() function (by legacy) returns with a reference.  So if it is returns with nullptr, then the reference (to assigned) would be nullptr as well, but later I cant be change.

Second problem: For the dfu operation both parameters (source and target)  can be  distributed file . The source can have and if it has then I should copy it to target. The target shouldn't already have History (because it will be a new file) but from now it should have to store this operation. 

Practically I should write history for the target file (if it is distributed) from DFU filecopy.